src/fauxtoshop.cpp

```diff
@@ -1,517 +1,492 @@
+ * Usage: Opens an image in the library and add to the window
+ * Params: img (GBufferedImage), gw (GWindow)
+ * ------------------------------------------------------------------------------------------------
+ * Returns: None. Void function
+*/
+void openImage(GBufferedImage &img, GWindow &gw) {
+    // adding a BufferedImage to GWindow
+    gw.setCanvasSize(img.getWidth(), img.getHeight());
+    gw.add(&img, 0, 0);
+}
+
+/*
+ * Function: selectFilterOptions
+ * Usage: Prompts the user to select 1 of 4 filter options as an integer
+ * Params: None
+ * ------------------------------------------------------------------------------------------------
+ * Returns: optionNum - the option as an Integer
+*/
+int selectFilterOptions() {
+    //Give user 4 filter options: scatter, edge detection, green screen, compare with other image
+    cout << "Which image filter would you like to apply?" << endl;
+    cout << "        1. Scatter" << endl;
+    cout << "        2. Edge detection" << endl;
+    cout << "        3. 'Green screen' with another image" << endl;
+    cout << "        4. Compare image with another image " << endl;
+    // Validate that the user is entering an integer data type between 1 and 4
+    return getValidInteger("Enter option value between 1 and 4:", 1, 4);
+}
+
+/*
+ * Function: getValidInteger
+ * Usage: Ensure user integer input is valid between a range, other reprompt the user to re-enter.
+ * Params: promptMessage (string), lowerBound (integer), upperBound (integer - default INT_MAX)
+ * ------------------------------------------------------------------------------------------------
+ * Returns: userInput (Integer)
+*/
+int getValidInteger(const string& promptMessage, int lowerBound, int upperBound = INT_MAX) {
+    while (true) {
+        int userInput = getInteger(promptMessage);
+        if (userInput >= lowerBound && userInput <= upperBound ) {
+            return userInput;
+        }
+        cout << "Invalid input! Please try again" << endl;
+    }
+}
+
+/*
+ * Function: validateLoctionInput
+ * Usage: Ensure user input for specifying a location on the canvas/ window is valid,
+ * otherwise reprompt the user to re-enter
+ * The user can enter the location in '(<row>, <col>)' format or hit enter
+ * to do this with a mouse click
+ * Params: row (integer), col (integer)
+ * ------------------------------------------------------------------------------------------------
+ * Returns: nothing. Void function
+*/
+void validateLocationInput(int& row, int& col) {
+    while (true) {
+        string locationInput = getLine(
+                    "Enter sticker location as '(<row>, <col>)'or hit enter to use the mouse: ");
+        // Validate that the input starts and ends with "("")" and contains ","
+        if (locationInput.empty()) {
+            cout << "Click on the canvas to place your sticker :-)" << endl;
+            getMouseClickLocation(row, col);
+            break;
+        } else if (startsWith(locationInput,"(")
```

Comment:
nit: add a space between the parameters. I.e. `locationInput, "("` instead of `locationInput,"("`. Same below